Source/UI/PaymentForm/ViewModel/DefaultPaymentViewModel.swift

```diff
@@ -154,14 +154,15 @@ extension DefaultPaymentViewModel: PaymentViewControllerDelegate {
     validateMandatoryInputProvided()
   }
 
-  func payButtonIsPressed() {
-    guard let card = cardDetails.getCard() else { return }
-    isLoading = true
-    checkoutAPIService.createToken(.card(card)) { [weak self] result in
-      self?.isLoading = false
-      self?.cardTokenRequested?(result)
+    func payButtonIsPressed() {
+        guard let card = cardDetails.getCard() else { return }
+        logger.log(.paymentFormSubmitted)
```

Comment:
Should we move this line to ( line 158 instead ) first thing in the func payButtonIsPressed so it always gets logged when user taps on button ?

Reply:
Problem is if it fails the `guard`, **payment form hasn't been submitted**, so it would be an inconsistency in the log?

Reply:
As mentioned on other comment thread, I opened PR #320 to trigger a warn event if the button press was to fail silently (from UI perspective)